data-prepper-plugins/type-conversion-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/typeconversion/TypeConversionProcessor.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.typeconversion;
+
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.processor.AbstractProcessor;
+import org.opensearch.dataprepper.model.processor.Processor;
+import org.opensearch.dataprepper.model.record.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+@DataPrepperPlugin(name = "type_conversion", pluginType = Processor.class, pluginConfigurationType = TypeConversionProcessorConfig.class)
```

Comment:
This is somewhat similar to our mutate processors. So I think it should follow a similar naming convention. It should probably have a verb like those. And maybe should include "entries."

Some options:
* `convert_entry`
* `convert`